### ISSUE TITLE
The Nesting rule should be optional since it is widely used in Swift 4

### ISSUE THREAD

**Issue by swordray** [realm/SwiftLint#2416]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The nesting structure is widely used in Swift 4. It should be considered as a good design. For example:

```swift
Array.Iterator
Date.ReferenceType
Dictionary.Keys
Double.Exponent
IndexPath.Index
Int.Words
NSAttributedString.Key
NSLayoutConstraint.AxisKeys
Set.Index
String.Encoding
UIActivity.ActivityType
UIApplication.LaunchOptionsKey
UINavigationItem.LargeTitleDisplayMode
UITableView.Style
UITableViewCell.AccessoryType
UIView.AutoresizingMask
URL.ReferenceType
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?

0.27.0

* Installation method used (Homebrew, CocoaPods, building from source, etc)?

CocoaPods

* Paste your configuration file:

```yml
excluded:
- Pods
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.

No

* Which Xcode version are you using (check `xcode-select -p`)?

10.0

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
class UITableView { 
    enum Style {} 
}
```

**Comment by marcelofabri** [comment]
The nesting rule already allows defining a type within 1 level of nesting by default (your example doesn't trigger the rule). 

It's also customizable so you can change the warning threshold.

I'm closing this because of that, feel free to reopen if needed.

**Comment by swordray** [comment]
@marcelofabri you are right.
